velox/functions/lib/string/StringImpl.h

```diff
@@ -147,29 +147,50 @@ FOLLY_ALWAYS_INLINE int32_t charToCodePoint(const T& inputString) {
   return codePoint;
 }
 
+/// Invokes 'callable' for each code point in 'input'. Stops processing if
+/// 'callable' returns false. Throws if input is not a valid UTF-8 string.
+///
+/// @param input Valid UTF-8 string.
+/// @param callable A function taking a 'codePoint' integer and returning a
+/// boolean.
+/// @return The index of the start of the last codePoint successfully processed
+/// by 'callable' (i.e. 'callable' returned true).
+template <typename TString, typename Callable>
+int32_t applyToCodePoints(const TString& input, Callable callable) {
```

Comment:
out of completeness: it would be super nice to have unit tests testing this function's API.

Reply:
@mbasmanova